.github/actions/cargo-risczero-install/action.yaml

```diff
@@ -30,47 +30,49 @@ inputs:
+      shell: bash
+      run: |
+        if [ ! -z "${{ inputs.version }}" ] && [ ! -z "${{ inputs.ref }}" ]; then
+          echo "Error: Cannot specify both 'version' and 'ref' parameters"
+          exit 1
+        fi
+        if [ -z "${{ inputs.version }}" ] && [ -z "${{ inputs.ref }}" ]; then
+          echo "Error: Must specify either 'version' or 'ref' parameter"
+          exit 1
+        fi
 
-      # Only run these steps if using ref (building from source)
-      - name: checkout risc0
-        if: inputs.ref != ''
-        uses: actions/checkout@v4
-        with:
-          repository: 'risc0/risc0'
-          path: 'tmp/risc0'
-          ref: ${{ inputs.ref }}
-          lfs: true
-      - name: install cargo-risczero from git
-        if: inputs.ref != ''
-        run: cargo install --locked --path risc0/cargo-risczero --no-default-features --features "${{ inputs.features }}"
-        working-directory: tmp/risc0
-        shell: bash
-      - name: install r0vm from git
-        if: inputs.ref != ''
-        run: cargo install --locked --bin r0vm --path risc0/cargo-risczero --features "${{ inputs.features }}"
-        shell: bash
-        working-directory: tmp/risc0
+    # Only run these steps if using ref (building from source)
+    - name: checkout risc0
+      if: inputs.ref != ''
+      uses: actions/checkout@v4
+      with:
+        repository: 'risc0/risc0'
+        path: 'tmp/risc0'
+        ref: ${{ inputs.ref }}
+        lfs: true
+    - name: install cargo-risczero from git
+      if: inputs.ref != ''
+      run: cargo install --locked --path risc0/cargo-risczero --no-default-features --features "${{ inputs.features }}"
+      working-directory: tmp/risc0
+      shell: bash
+    - name: install r0vm from git
+      if: inputs.ref != ''
+      run: cargo install --locked --bin r0vm --path risc0/cargo-risczero --features "${{ inputs.features }}"
+      shell: bash
+      working-directory: tmp/risc0
 
-      # Only run this step if using version (installing from crates.io)
-      - name: install cargo-risczero from crates.io
-        if: inputs.version != ''
-        run: cargo install --locked cargo-risczero --version "${{ inputs.version }}" --no-default-features --features "${{ inputs.features }}"
-        shell: bash
+    # Only run this step if using version (installing from crates.io)
+    - name: install cargo-risczero from crates.io
+      if: inputs.version != ''
+      run: cargo install --locked cargo-risczero --version "${{ inputs.version }}" --no-default-features --features "${{ inputs.features }}"
+      shell: bash
 
-      - name: install toolchains
-        run: cargo risczero install ${{ inputs.toolchain-version != '' && format('--version {0}', inputs.toolchain-version) || '' }}
-        shell: bash
-      - name: cleanup
-        run: rm -rf tmp
-        shell: bash
+    - name: install rust toolchains
+      run: cargo run --bin rzup -- --verbose install rust ${{ inputs.toolchain-version }}
```

Comment:
`cargo run --bin rzup` will only work when `ref != ''` (i.e. when a ref is specified and not a version). When a version is specified, I'm not sure what the best approach is. Ideally, we could install `rzup` via curl-to-bash or from the latest release on `crates.io` and run `rzup install ${{ version }}` but that is not actually a valid command and I'm not immediately sure what the long-form version of this is.